### PR TITLE
[WC-1337]: Fix barcode-scanner

### PR DESCRIPTION
**Pull request by iobuhov**
## Checklist

-   Contains unit tests ✅
-   Contains breaking changes ❌
-   Contains Atlas changes ❌
-   Compatible with: 9️⃣
-   Did you update version and changelog? ✅
-   PR title properly formatted (`[XX-000]: description`)? ✅

## This PR contains

-   [x] Bug fix
-   [ ] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?

- Fix issue with camera resource not disposed on unmount

## Relevant changes

- Now, we controlling stream explicitly to make sure it disposed after use
- add `stopAdyncDecode` coll to make sure we don't have setTimeout loops

### CODE REVIEW

File: packages/pluggableWidgets/barcode-scanner-web/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+### Fixed
+
+-   We fixed issue with camera, when it keep capturing video after widget is closed.
```
Comment (HedwigAR) [inline packages/pluggableWidgets/barcode-scanner-web/CHANGELOG.md:11]:
```suggestion
-   We fixed an issue where the camera kept capturing after the widget was closed.
```

**Review by HedwigAR — COMMENTED** [review]
I have one suggestion

**Review by HedwigAR — APPROVED** [review]
Looks good to me

**Review by rahmanunver — DISMISSED** [review]

File: packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCustomErrorMessage.ts [hunk 2]
```diff
@@ -1,27 +1,15 @@
...
 type ErrorCb<E extends Error = Error> = (error: E) => void;
 
 type UseCustomErrorMessageHook = () => [string | null, ErrorCb];
 
 function getErrorMessage<E extends Error>(error: E): string | null {
-    // Error comes from `zxing`
-    if (error instanceof Exception) {
-        return "Error in barcode scanner: an unexpected error occurred while detecting a barcode in the camera media stream.";
+    if (error instanceof Error) {
+        return `Error in barcode scanner: ${error.message}`;
```
Comment (rahmanunver) [inline packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCustomErrorMessage.ts:9]: I assume error messages are clear enough that we don't need to return a string explaining it. Other than that case I don't see anything else.

Comment (iobuhov) [reply packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCustomErrorMessage.ts:9]: Yeah, that's probably worth to remove.

Comment (iobuhov) [reply packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCustomErrorMessage.ts]: @rahmanunver , done

**Review by rahmanunver — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**